### DISCUSSION TITLE
ahk.win_move to another monitor

### DISCUSSION THREAD

**Discussion by trajano** [spyoungtech/ahk#322] [Q&A] [answered]
Hi have two monitors, I would like to resize the window so it is in `0,0` of another monitor.
I am thinking maybe `ahk.win_move` should add a `monitor` parameter to determine that since the monitor computation may not be trivial (i.e. if I have one monitor landscape, another portrait and say there's a laptop screen added to the mix as well.

What I did at the moment was use *AutoHotkey Window Spy* to copy the X,Y values from there.

**Comment by spyoungtech** [comment]
> ahk.win_move should add a monitor parameter

Hmm. Yeah. Similar ideas have been discussed at some length in #85 -- though, I'm not 100% confident in my ability to correctly deal with all monitor scenarios and how reliable that could be. So, for now, the exercise is left to the user, as is the case in AutoHotkey. If AutoHotkey adds an API for this, it will be added in the wrapper.

But if someone is motivated to create the functionality, it could perhaps be a good project for an [extension package](https://ahk.readthedocs.io/en/latest/extending.html). And there's always room for successful extension projects to be featured in the readme  :)

Something I do want to provide in the future is a rough analog to Window Spy, which may also be another extension use case.

**Reply by trajano** [reply]
Using AHK's WinMove likely won't work see https://www.autohotkey.com/boards/viewtopic.php?t=17955

I know I put in the values myself to get it working but it does not appear to be working correctly.  x=-3840 on dual 4K displays doesn't put the window on the left side of the screen as I would expect.

**Reply by spyoungtech** [reply]
For whatever it's worth, I made a small utility package to get monitor information called [`wmutil`](https://github.com/spyoungtech/wmutil). It _should_ (at least in my initial testing) supply the correct monitor positions which can be used with `win.move`.

It also has some utilities that work with getting the monitor for a particular window or at a particular point, using [MonitorFromWindow](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-monitorfromwindow) or [MonitorFromPoint](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-monitorfrompoint) respectively.

**Comment by trajano** [comment, marked as answer]
I used the `screeninfo` library and the raw Win32 API to move the window

```python
import screeninfo
import ctypes
from ctypes import wintypes

user32 = ctypes.WinDLL('user32', use_last_error=True)

def monitor_xy(x: int, y: int, monitor=0) -> tuple[int, int]:
    monitors = screeninfo.get_monitors()
    if monitor >= len(monitors):
        raise ValueError(
            f"Monitor index {monitor} is out of range. There are only {len(monitors)} monitors."
        )
    return (x + monitors[monitor].x, y + monitors[monitor].y)

def one_note() -> None:
    one_note_ahk_title = "ahk_exe ONENOTE.EXE"
    (x, y) = monitor_xy(0, 0, monitor=1)
    window =ahk.win_get(one_note_ahk_title)
        
    if not user32.MoveWindow(int(window.id), x, y, 1920, 2100, True):
        raise ctypes.WinError(ctypes.get_last_error())
    ahk.win_activate(one_note_ahk_title)
```
